plugin/trino-hive/src/main/java/io/trino/plugin/hive/cache/HiveCacheKeyProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.cache;
+
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.cache.CacheKeyProvider;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public final class HiveCacheKeyProvider
+        implements CacheKeyProvider
+{
+    @Override
+    public Optional<String> getCacheKey(TrinoInputFile inputFile)
+            throws IOException
+    {
+        // We use file length in the cache key instead of lastModified to avoid making extra FS calls to fetch last modified time of file.
+        // We make a simplifying assumption that files are not rewritten with exactly same path and size but different content.
```

Comment:
I'm not sure about this assumption

Reply:
The alternative was to extend TrinoFileSystem API to add `TrinoInputFile newInputFile(Location location, long length, Instant lastModified)` which allows us to reuse the file modified time from HiveSplit instead of making an FS call. But this makes the TrinoFileSystem API a bit more complicated.
I discussed with @electrum that we can make this simplification since caching is an opt-in feature and someone modifying files in-place can turn it off. We can also make a config flag to use last modified time in cache key if we find that this turns out to be problematic for some users.
One can also migrate hive tables to delta/iceberg to avoid this issue entirely.

Reply:
But it would potentially lead into a case where we might read stale file untill - ttl has been expired - unlike the previous approach where we would update the cache once we see a new file

Reply:
`lastModified` from HiveSplit could be an option (I do agree it would be complex)  as it would allow us to unify the caching logic for both the `CachingDirectoryLister` and the Alluxio cache as well.

Reply:
> potentially lead into a case where we might read stale file untill - ttl has been expired

I think that would only happen for a file that is rewritten in-place with exactly same size but different content. Modifying files in-place is not a recommended approach even in hive tables. Trino definitely doesn't do that in any of it's write paths. It's not clear to me that we should add complexity to handle this corner case until we know that is something that users actually need.

Reply:
I've changed the approach to use `lastModified` from HiveSplit